app/scripts/views/sign_up.js

```diff
@@ -51,6 +51,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, Session, AuthErrors,
         return p(false);
       }
 
+      /**
+       * check if there is an account we can suggest for an OAuth flow.
+       */
+      var account = this.user.getChooserAccount();
+      var serviceName = this.relier.get('serviceName');
+
+      if (account.get('email') && serviceName) {
```

Comment:
Ok thanks for feedback. will update. As I was saying in IRC, this changes this flow a bit: 
https://dxr.mozilla.org/mozilla-central/source/services/fxaccounts/FxAccountsOAuthClient.jsm#38